frontend/src/components/Dropzone.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback } from 'react';
+import { useDropzone } from 'react-dropzone';
+import { parse, ParseConfig } from 'papaparse';
+import Box from '@mui/material/Box';
+import FileUploadIcon from '@mui/icons-material/FileUpload';
+import { Typography } from '@mui/material';
+import { Stack } from '@mui/material';
+
+export interface DropzonePureProps {
+  isDragActive: boolean;
+  getRootProps: any;
+  getInputProps: any;
+}
+
+export function DropzonePure({
+  getRootProps,
+  getInputProps,
+  isDragActive,
+}: DropzonePureProps) {
+  return (
+    <Box
+      component={Stack}
+      direction="column"
+      justifyContent="center"
+      sx={{
+        width: 700,
+        height: 400,
+        backgroundColor: '#E2EAE6',
+        m: 'auto',
+        mt: 8,
+        border: '3px dashed',
+        borderColor: 'primary.main',
+      }}
+      {...getRootProps()}
+    >
+      <Typography align="center">
+        {' '}
+        <FileUploadIcon color="primary" sx={{ fontSize: 80 }} />{' '}
+      </Typography>
+      <input {...getInputProps()} />
+      {isDragActive ? (
+        <Typography color="primary" align="center">
+          Drop file here...
+        </Typography>
+      ) : (
+        <Typography color="primary" align="center">
+          <b>Choose a CSV file</b> or drag it here.
+        </Typography>
+      )}
+    </Box>
+  );
+}
+
+export default function Dropzone() {
+  // @ts-ignore
+  const onDrop = useCallback((acceptedFiles) => {
+    acceptedFiles.forEach((file: File) => {
+      if (file.type !== 'text/csv') {
+        // Not the right type of file, so we skip it for now.
+        alert(
+          'Only .csv files are supported. Uploaded file is called:' +
+            file.name +
+            ':'
```

Comment:
We don't need it, per say, but @illume had suggested using ":" as a way to explicitly distinguish the end of a file name. I wonder if he can provide a better explanation?

Reply:
Maybe it's better to remove it. It's a bit confusing.

Yeah, the goal is to make sure it's easy to see the full file name when there are things like spaces in there (because there's no formatting in alert). However, mostly this won't matter I think.